runtime/cudaq/operators/boson_op.cpp

```diff
@@ -43,20 +47,33 @@ std::string boson_handler::op_code_to_string() const {
     else
       str += "(N" + std::to_string(offset) + ")";
   }
-  for (auto i = 0; i < this->additional_terms; ++i)
+  if (this->additional_terms > 0) {
     str += "Ad";
-  for (auto i = 0; i > this->additional_terms; --i)
+    if (this->additional_terms > 1)
+      str += "^" + std::to_string(this->additional_terms);
+  } else if (this->additional_terms < 0) {
     str += "A";
+    if (-this->additional_terms > 1)
+      str += "^" + std::to_string(-this->additional_terms);
+  }
   return std::move(str);
 }
 
+// used internally for canonical evaluation -
+// use a encoding that makes it convenient to reconstruct the operator
```

Comment:
another option would be to use an integer encoding instead, similar to how we use doubles for the (currently only spin ops) serialization format